### PR TITLE
AuthN: Change the external service account state on plugin state change

### PR DESCRIPTION
**Pull request by gamab**
**What is this feature?**

This PR adds a listener to plugin state changes to enable/disable their associated service account accordingly.

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

https://github.com/grafana/grafana-enterprise/issues/5824

**Special notes for your reviewer:**

I think I'll need to disable external service accounts in the plugin initializer step by default.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/serviceaccounts/manager/store.go [hunk 1]
```diff
@@ -8,18 +8,19 @@ import (
...
-	RetrieveServiceAccount(ctx context.Context, orgID, serviceAccountID int64) (*serviceaccounts.ServiceAccountProfileDTO, error)
-	RetrieveServiceAccountIdByName(ctx context.Context, orgID int64, name string) (int64, error)
 	DeleteServiceAccount(ctx context.Context, orgID, serviceAccountID int64) error
-	MigrateApiKeysToServiceAccounts(ctx context.Context, orgID int64) (*serviceaccounts.MigrationResult, error)
-	MigrateApiKey(ctx context.Context, orgID int64, keyId int64) error
-	ListTokens(ctx context.Context, query *serviceaccounts.GetSATokensQuery) ([]apikey.APIKey, error)
-	RevokeServiceAccountToken(ctx context.Context, orgId, serviceAccountId, tokenId int64) error
-	AddServiceAccountToken(ctx context.Context, serviceAccountID int64, cmd *serviceaccounts.AddServiceAccountTokenCommand) (*apikey.APIKey, error)
 	DeleteServiceAccountToken(ctx context.Context, orgID, serviceAccountID, tokenID int64) error
+	EnableServiceAccount(ctx context.Context, orgID, serviceAccountID int64, enable bool) error
```
Comment (gamab) [inline pkg/services/serviceaccounts/manager/store.go:15]: Only new entry, the rest is re-ordering

File: pkg/services/serviceaccounts/database/store_test.go [hunk 2]
```diff
@@ -495,3 +495,62 @@ func TestServiceAccountsStoreImpl_SearchOrgServiceAccounts(t *testing.T) {
...
+		},
+	}
+	for _, tc := range tt {
+		t.Run(tc.desc, func(t *testing.T) {
+			ctx := context.Background()
+
+			err := store.EnableServiceAccount(ctx, orgID, tc.id, tc.enable)
+			require.NoError(t, err)
+
+			switch tc.isUser {
```
[the remark this answers is not in the document]
Comment (gamab) [reply pkg/services/serviceaccounts/database/store_test.go]: I changed the test to make it liner but also check for no collateral update

File: pkg/services/serviceaccounts/database/store_test.go [hunk 3]
```diff
@@ -495,3 +495,67 @@ func TestServiceAccountsStoreImpl_SearchOrgServiceAccounts(t *testing.T) {
...
+		{
+			desc:       "should not disable user",
+			id:         3,
+			enable:     false,
+			wantStates: map[int64]bool{1: true, 2: true, 3: true},
+		},
+	}
+	for _, tc := range tt {
+		t.Run(tc.desc, func(t *testing.T) {
+
```
Comment (gamab) [inline pkg/services/serviceaccounts/database/store_test.go:554]:
```suggestion
```

File: pkg/services/serviceaccounts/database/store_test.go [hunk 4]
```diff
@@ -495,3 +495,66 @@ func TestServiceAccountsStoreImpl_SearchOrgServiceAccounts(t *testing.T) {
...
 	}
 }
+
+func TestServiceAccountsStoreImpl_EnableServiceAccounts(t *testing.T) {
+	ctx := context.Background()
+
+	initUsers := []tests.TestUser{
+		{Name: "satest-1", Role: string(org.RoleViewer), Login: "sa-satest-1", IsServiceAccount: true},
+		{Name: "satest-3", Role: string(org.RoleEditor), Login: "sa-satest-3", IsServiceAccount: true},
+		{Name: "usertest-2", Role: string(org.RoleEditor), Login: "usertest-2", IsServiceAccount: false},
```
Comment (gamab) [inline pkg/services/serviceaccounts/database/store_test.go:505]:
```suggestion
		{Name: "satest-2", Role: string(org.RoleEditor), Login: "sa-satest-2", IsServiceAccount: true},
		{Name: "usertest-3", Role: string(org.RoleEditor), Login: "usertest-3", IsServiceAccount: false},
```

File: pkg/services/serviceaccounts/database/store_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (linoman) [reply pkg/services/serviceaccounts/database/store_test.go]: The changes look great!

**Review by linoman — APPROVED** [review]
This LGTM! Great job!

**Review by andresmgot — APPROVED** [review]
LGTM :+1:

**Pull request merged**